### PR TITLE
Refactor StreamReader::startStripe to take the Stripe directly

### PR DESCRIPTION
**Pull request by HuamengJiang**
This PR is in preparation for dictionary sharing implementation in 
DWRF format, a file format level feature that allows
multiple keys in a flat map to share the same dictionary data stream.
This PR is refactor only and contains no logic change.

Test plan - existing unit tests. We sampled the top 3 namespaces for 
tables to run integration tests on and the result is clean.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-orc/src/main/java/com/facebook/presto/orc/AbstractOrcRecordReader.java [hunk 1]
```diff
@@ -662,7 +662,7 @@ private void advanceToNextStripe()
             Map<Integer, ColumnEncoding> columnEncodings = stripe.getColumnEncodings();
             for (StreamReader column : streamReaders) {
                 if (column != null) {
-                    column.startStripe(dictionaryStreamSources, columnEncodings);
```
Comment (arunthirupathi) [inline presto-orc/src/main/java/com/facebook/presto/orc/AbstractOrcRecordReader.java:665]: remove the 2 local fields dictionaryStreamSources and columnEncodings.

**Review by arunthirupathi — APPROVED** [review]
This looks good once the comment is addressed. Please run the validation service

**Comment by sdruzkin** [comment]
LGTM

**Review by sdruzkin — APPROVED** [review]
LGTM

**Review by highker — COMMENTED** [review]
Could you make your commit message bound within 72 characters per line? Also, let's remove any description that is open-source-friendly. cc: @arunthirupathi

**Review by highker — APPROVED** [review]
otherwise LGTM

**Comment by HuamengJiang** [comment]
> otherwise LGTM

Updated the summary to wrap lines and remove non-OSS content. :D

**Comment by arunthirupathi** [comment]
Change the Title to "Refactor the startStripe to take Stripe directly"

Change the summary to: (Fit this to 72 column length)
This PR is in preparation for dictionary sharing 
implementation in DWRF format.  This PR has no 
logic change and is only refactoring.

**Comment by highker** [comment]
@HuamengJiang, you need to amend the commit message from you local env and force push so that git master can catch it :D

**Pull request merged**